core/actions/operation.ts

```diff
@@ -71,8 +71,26 @@ export class Operation extends ActionBuilder<dataform.Operation> {
   // We delay contextification until the final compile step, so hold these here for now.
   private contextableQueries: Contextable<ICommonContext, string | string[]>;
 
-  constructor(session?: Session) {
+  constructor(session?: Session, config?: dataform.ActionConfig) {
     super(session);
+    if (!session || !config) {
```

Comment:
if one is set and the other is not, the set one will just be ignored? that seems misleading

Reply:
If either are unset, then the constructor returns early, because all of the methods in the constructor will throw otherwise.

Longer term the plan is to make `config` private, and require session and config to be defined - but for backwards compatibility with the existing Javascript API session methods (e.g. https://github.com/dataform-co/dataform/blob/cab21959a1857f964b113e94d9a2d3b8fbc51c0a/core/session.ts#L280) I've allowed them to be optional.